### ISSUE TITLE
False positives in angular template

### ISSUE THREAD

**Issue by LeoVie** [schoero/eslint-plugin-better-tailwindcss#177]
Using the config from https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/parsers/angular.md, eslint reports some false positives in this template:

```html
<div [class]="{
  'hidden': renderMode() === 'popover',
}"></div>
<div [class]="{
  'lg:rounded-r-lg': renderMode() === 'modal',
  'rounded-r-lg': renderMode() === 'popover',
}"></div>
```

In line 2, it's reporting 
```
Unregistered class detected: popover (better-tailwindcss/no-unregistered-classes
``` 

and in line 5, it's reporting 
```
Incorrect class order. Expected

:rounded-r-lg'

to be

rounded-r-lg'
(better-tailwindcss/enforce-consistent-class-order)
ESLint:

Unnecessary whitespace. Expected

:rounded-r-lg'

to be

rounded-r-lg'
(better-tailwindcss/no-unnecessary-whitespace)
```

Switching the classes in line 5 and 6 makes the second error disappear. So I think, the detection of wrong order is correct, but applying the automatic fix is destroying the template to

```html
...
<div [class]="{
  'lgrounded-r-lg': renderMode() === 'modal',
  'rounded-r-lg': renderMode() === 'popover',
}"></div>
```

It's possible to workaround that false positives by moving the equality check into the typescript part of the component and ordering the classes manually
```html
<div [class]="{
  'hidden': isPopover(),
}"></div>
<div [class]="{
  'rounded-r-lg': isPopover(),
  'lg:rounded-r-lg': isModal(),
}"></div>
```

**Comment by schoero** [comment]
Thanks for reporting.

I found a few additional issues with angular and I'm working on fixes on #182.

Unfortunately the angular parser is very hard to work with because it is completely different from the other parsers and it is very hard to gather the required information from the angular AST. 🥲

I hope to a few hours this weekend to fix those things. 🤞

**Comment by yossiyedid** [comment]
Hi @schoero any progress with this one? 
looks like the plugin can parse angular template as follow
`<some-component [class]="someFunction({prop: value}" />`
